envoy/http/filter.h

```diff
@@ -727,6 +706,22 @@ class StreamDecoderFilterCallbacks : public virtual StreamFilterCallbacks,
    */
   virtual void encodeMetadata(MetadataMapPtr&& metadata_map) PURE;
 
+  /**
```

Comment:
This part is just to revert the code back to before. i.e., deprecate `DecoderFilterWatermarkCallbacks`